cmake/secp256k1.cmake

```diff
@@ -9,6 +9,7 @@
 enable_language(C)
 set(CMAKE_C_STANDARD 90)
 set(CMAKE_C_EXTENSIONS OFF)
+string(APPEND CMAKE_C_COMPILE_OBJECT " ${APPEND_CPPFLAGS} ${APPEND_CFLAGS}")
```

Comment:
Why are we only setting this for this specific target? Seems a bit brittle to have to remember to add this in case a new C target is added.

Reply:
> Why are we only setting this for this specific target?

I'd rather not see that in this code :)

The `CMAKE_C_COMPILE_OBJECT` variable is modified as soon as it becomes available, which is just after enabling the C language.

No C target can be added before enabling the C language. Therefore, when it happens there are two options. Either it will be add after this code or this code will be moved to ensure the correct order of language enabling.